fetch.ts

```diff
@@ -17,6 +17,83 @@ export function initFetch() {
   });
 }
 
+export class Headers {
+  private map: Map<string, string>;
+
+  constructor(
+    headers?: Headers | Array<[string, string]> | { [name: string]: string }
+  ) {
+    if (!headers) {
+      this.map = new Map();
+    } else if (headers instanceof Headers) {
+      const arr = [];
+      for (const header of headers) {
+        arr.push(header);
+      }
+      this.map = new Map(arr);
+    } else if (Array.isArray(headers)) {
+      this.map = new Map(headers);
+    } else {
+      this.map = new Map();
+      const names = Object.keys(headers);
+      names.forEach(name => {
+        this.map.set(name, headers[name]);
+      });
+    }
+  }
+
+  append(name: string, value: string): void {
+    if (this.map.has(name)) {
+      const oldValue = this.map.get(name);
+      this.map.set(name, oldValue + ", " + value);
+    } else {
+      this.map.set(name, value);
+    }
+  }
+
+  delete(name: string): void {
+    this.map.delete(name);
+  }
+
+  entries(): IterableIterator<[string, string]> {
+    return this.map.entries();
+  }
+
+  get(name: string): string | null {
+    if (this.map.has(name)) {
+      return this.map.get(name).split(", ")[0];
+    }
+    return null;
+  }
+
+  getAll(name: string): string[] {
+    const value = this.map.get(name);
+    const arr = value ? value.split(", ") : [];
+
+    return arr;
+  }
```

Comment:
How about remove 'get' but just provide 'getAll'?
```
get(name: string): string | null {
  return this.map.get(name);
}
```

Reply:
I think I should remove 'getAll', because 'getAll' is obsolete.  It is only work in IE. 
Here is link.
[https://developer.mozilla.org/en-US/docs/Web/API/Headers/getAll](https://developer.mozilla.org/en-US/docs/Web/API/Headers/getAll)

Reply:
It makes sense.
I forgot that the APIs should be browser compatible.